packages/web/src/common/store/pages/saved/sagas.js

```diff
@@ -76,15 +76,18 @@ function* watchFetchSaves() {
           save_item_id: save.track.track_id
         }))
 
-        const fullSaves = newTablesEnabled
-          ? Array(account.track_save_count)
-              .fill(0)
-              .map((_) => ({}))
-          : saves
+        const fullSaves = Array(account.track_save_count)
+          .fill(0)
+          .map((_) => ({}))
 
         fullSaves.splice(offset, saves.length, ...saves)
 
-        yield put(actions.fetchSavesSucceeded(fullSaves))
+        yield put(
+          actions.fetchSavesSucceeded(newTablesEnabled ? fullSaves : saves)
+        )
+        if (limit && saves.length < limit) {
```

Comment:
Should this be like this to match the one below?

```suggestion
        if (limit > 0 && saves.length < limit) {
```